besu/src/main/java/org/hyperledger/besu/cli/custom/RpcAuthFileValidator.java

```diff
@@ -97,7 +99,20 @@ private static boolean verifyAllEntriesHaveValues(final TomlParseResult tomlPars
         .dottedKeySet()
         .parallelStream()
         .filter(keySet -> !keySet.contains("password"))
-        .allMatch(dottedKey -> verifyArray(dottedKey, tomlParseResult));
+        .allMatch(dottedKey -> verifyEntry(tomlParseResult, dottedKey));
+  }
+
+  private static boolean verifyEntry(final TomlParseResult tomlParseResult, final String key) {
+    if (key.endsWith(PRIVACY_PUBLIC_KEY)) {
+      return verifyString(key, tomlParseResult);
+    } else {
+      return verifyArray(key, tomlParseResult);
+    }
+  }
+
+  private static boolean verifyString(final String key, final TomlParseResult tomlParseResult) {
+    final String value = tomlParseResult.getString(key);
+    return tomlParseResult.isString(key) && value != null && !value.isEmpty();
```

Comment:
Based on the source code of `isString()` you'll want to perform the null check before invoking `tomlParseResult.isString(key)`

Reply:
The reason is the source version of that method for me comes from `TomlTable`, where there is the require not null check 😉 
```  
default boolean isString(String dottedKey) {
    requireNonNull(dottedKey);
    return isString(parseDottedKey(dottedKey));
  }
```

Reply:
Nevermind, my bad, I misread the `key` as `value` 🙄